llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -281,25 +281,50 @@ Expected<PaddingRef> PaddingRef::get(Expected<MCObjectProxy> Ref) {
   return PaddingRef(*Specific);
 }
 
-static void writeRelocations(ArrayRef<MachO::any_relocation_info> Rels,
-                             SmallVectorImpl<char> &Data) {
-  for (auto Rel : Rels) {
+static void writeRelocationsAndAddends(
+    ArrayRef<MachO::any_relocation_info> Rels,
+    ArrayRef<MachObjectWriter::AddendsSizeAndOffset> Addends,
+    SmallVectorImpl<char> &Data) {
+  writeVBR8(Rels.size(), Data);
+  for (unsigned I = 0; I < Rels.size(); I++) {
     // FIXME: Might be better just encode raw data?
-    writeVBR8(Rel.r_word0, Data);
-    writeVBR8(Rel.r_word1, Data);
+    writeVBR8(Rels[I].r_word0, Data);
+    writeVBR8(Rels[I].r_word1, Data);
+  }
+  writeVBR8(Addends.size(), Data);
+  for (unsigned I = 0; I < Addends.size(); I++) {
+    writeVBR8(Addends[I].Value, Data);
```

Comment:
Ah but you're storing them in VBR encoding anyway....

Reply:
Yes, it is more space efficient that way